### PR TITLE
Add document_collection schema

### PR DESCRIPTION
**Pull request by fofr**
Details:
* Body is optional for collections
* First public at is required like other editioned content
* A document collection must have an ordered list of groups
  * Each group must have a title and an ordered list of documents, the group may have some body copy
* Collections can be political
* Collections can be withdrawn
* Include tags – these are not really used except for emails, see #292 for why they weren't removed from other formats

Links:
* Add documents GUID list, make this a required field
* Whitehall marks "policy areas" as being required in the editor, but content schemas say this is "largely deprecated", so it has been intentionally left out. (Policy areas are included by base links)
* Add topical events
* Lead organisations are also required, we've changed the pattern for organisations to "organisations" in links and emphasised organisations in details – there isn't a pattern yet for which bits must be required
to match current behaviour.

Include examples based on:
https://www.gov.uk/government/collections/national-driving-and-riding-standards
https://www.gov.uk/government/collections/financial-sanctions-regime-specific-consolidated-lists-and-releases

https://trello.com/c/PMeMQb5E/410-11-document-collection-migration-mvp-content-schema-examples-and-front-end-work

### CODE REVIEW

**Comment by fofr** [comment]
Government frontend WIP to render this format:
https://github.com/alphagov/government-frontend/compare/mvp-document-collections

**Comment by fofr** [comment]
Closing while adding more examples.

File: formats/publication/frontend/examples/political_publication.json [hunk 1]
```diff
@@ -71,16 +71,6 @@
         "locale": "en"
       }
     ],
-    "document_collections": [
```
Comment (boffbowsh) [inline formats/publication/frontend/examples/political_publication.json]: Why is this removed?

**Comment by fofr** [comment]
Test failures are unrelated.
